### PR TITLE
chore: full deploy script

### PR DESCRIPTION
**Pull request by jbrower95**
**Summary:**
- For local deploys / devnets, we want a script that can deploy onto a brand new network.
- This script is a start -- it deploys the core contracts and is going to be used by our devnet (https://github.com/Layr-Labs/layr-devnet)

**Run with:**
`RUST_LOG=forge,foundry=trace forge script script/deploy/local/Deploy_From_Scratch.s.sol --rpc-url $RPC_URL --private-key $PRIVATE_KEY --broadcast --sig "run(string memory configFile)" -- devnet/deploy_from_scratch.anvil.config.json`

### CODE REVIEW

File: script/deploy/local/Deploy_From_Scratch.s.sol [hunk 1]
```diff
@@ -0,0 +1,645 @@
...
+import "../../../src/test/mocks/ETHDepositMock.sol";
+
+import "forge-std/Script.sol";
+import "forge-std/Test.sol";
+
+// # To load the variables in the .env file
+// source .env
+
+// # To deploy and verify our contract
+// forge script script/deploy/devnet/Deploy_From_Scratch.s.sol --rpc-url $RPC_URL --private-key $PRIVATE_KEY --broadcast --sig "run(string memory configFile)" -- deploy_from_scratch.anvil.config.json
```
Comment (gpsanant) [inline script/deploy/local/Deploy_From_Scratch.s.sol:34]: edit comment

Comment (gpsanant) [reply script/deploy/local/Deploy_From_Scratch.s.sol:34]: forge script script/deploy/local/Deploy_From_Scratch.s.sol --rpc-url $RPC_URL --private-key $PRIVATE_KEY --broadcast --sig "run(string memory configFile)" -- devnet/deploy_from_scratch.anvil.config.json

Comment (gpsanant) [reply script/deploy/local/Deploy_From_Scratch.s.sol:34]: note that it doesn't do the strategy factory as well

**Review by gpsanant — APPROVED** [review]

File: script/deploy/local/Deploy_From_Scratch.s.sol (hunk 1 above)
Comment (jbrower95) [reply script/deploy/local/Deploy_From_Scratch.s.sol]: updated comment!

**Pull request merged**